### ISSUE TITLE
Changing learning rate and drop_out values of a model 'on the fly'.

### ISSUE THREAD

**Issue by faltinl** [rstudio/keras#1114]
In the past, methods for adaptive learning rates have been an active area of research going from continuously varying learning rate to the extreme of using just one very high constant learning rate to achieve what the authors call super convergence in [arXiv:1708.07120v3 ](https://arxiv.org/abs/1708.07120) - see this paper for further references, too. 

The objective in these cases most frequently is improving convergence in case of one given, but often extremely large data set. The case of rolling prediction or classification, however, poses similar problems even in case of much smaller data sets whose statistical properties may change considerably over time. Just including the newly arrived data into the existing data pool doesn't seem to be appropriate, since it may take an amount of approx. the same size as the already existing data to avoid discrimination of the new properties; i.e. the adaptive process is rather slow. 

A rolling prediction may require different values for several parameters like learning rate `LRate` in the optimizer, `drop_out` values in particular layers and or `model_weights` of the whole model. While it is perfectly simple to `set_weights` in a given model betwen two training cycles, to my knowledge, changing `LRate` requires application of the `compile()` command; changing `drop_out` values requires a complete read-in of the model and, then, the `compile()` command. The execution of the `compile()` command as well as reading-in of the whole network structure requires a relatively long time which may become prohibitive in case of more frequent adaptation of these parameters.

My question is: are there any options to shorten these procedures, e.g. by commands similar to `set_weights`? Thank you in advance for any suitable hints!
